src/solver/impls/petsc/petsc.cxx

```diff
@@ -118,10 +118,6 @@ int PetscSolver::init(int NOUT, BoutReal TIMESTEP) {
   nout = NOUT;
   tstep = TIMESTEP;
 
-  // Set the rhs solver function
-  // func = f;
-
-  PetscInt n2d = n2Dvars();       // Number of 2D variables
```

Comment:
I wonder if that fact that this isn't used is actually indicating a bug -- `n3d` is used to determine how many variables there are at a given grid point etc. but presumably if we have evolving 2d fields we should include these as well? Take the extreme of a 2D problem, I think the PETSc solver may behave in a strange way in that case.